### DISCUSSION TITLE
Why does installing homebrew require locally cloning the entire homebrew-core tap repository?

### DISCUSSION THREAD

**Discussion by richiksc** [Homebrew/discussions#231] [Everyday usage]
Today I installed homebrew into `/opt/homebrew` using the untar method of installation. When I tried to `brew install gradle --skip-dependencies`, it began to clone the entire homebrew-core repository. The repository is not a small repository and takes considerable time to clone and occupies 100s of megabytes of disk space.  Since you can't install the tools and applications (either building from source or installing a bottle) without an internet connection anyway, what's the point of keeping a local cache of all the formulas? Wouldn't it be a better user experience for the formula repository to just be online and then it fetches the latest copy of that specific formula when you `brew install` something?

**Comment by jonchang** [comment]
See https://github.com/Homebrew/discussions/discussions/226 and related discussions/issues.

**Reply by richiksc** [reply]
> See #226 and related discussions/issues.

That one was about full clone vs shallow clone. My question is, is it necessary to clone the tap repository at all?

**Reply by carlocab** [reply]
The `homebrew/core` tap contains the core formulae. I don't know how you'll be able to install anything without it. (I guess you could use someone else's tap.)

I suppose you could try to get a copy of the repository without cloning, but then keeping it updated is going to get annoying. You'll have to use some mechanism other than `brew update`.

Note that `brew install` can work without an internet connection if you have a cached copy of the bottle (or source tarball), so it is not true that you need an internet connection to install anything. (Though it is the most common circumstance under which you use `brew install`.)

Another reason is that querying the `homebrew/core` repo without cloning it every time you want to install something is more complicated to implement since it's a lot more than just vanilla git. If you have an implementation of such a feature, perhaps you could open a PR at Homebrew/brew so the maintainers can take a look.

**Comment by tddschn** [comment]
I'm wondering if it's possible to only install the brew command when installing Homebrew, without tapping `core` and `cask`?

If I want to install a formula later, I could just run `brew install <URL-for-that-formula>`.

**Reply by carlocab** [reply]
Try setting `HOMEBREW_INSTALL_FROM_API`. See `man brew` for details.
